pkg/proxy/register.go

```diff
@@ -38,24 +40,67 @@ func (p *Register) AddMany(routes []*Route) error {
 
 // Add register a new route
 func (p *Register) Add(route *Route) error {
-	return p.AddWithInOut(route, InChain{}, OutChain{})
-}
-
-// AddWithInOut register a new route with inbound/outbounds plugins
-func (p *Register) AddWithInOut(route *Route, inbound InChain, outbound OutChain) error {
 	definition := route.proxy
 
-	handler := p.proxy.Reverse(definition, inbound, outbound).ServeHTTP
+	p.params.Outbound = route.outbound
+	handler := &httputil.ReverseProxy{
+		Director:  p.createDirector(definition),
+		Transport: NewTransportWithParams(p.params),
+	}
+
 	matcher := router.NewListenPathMatcher()
 	if matcher.Match(definition.ListenPath) {
-		p.doRegister(matcher.Extract(definition.ListenPath), handler, definition.Methods, route.handlers)
+		p.doRegister(matcher.Extract(definition.ListenPath), handler.ServeHTTP, definition.Methods, route.inbound)
 	}
 
-	p.doRegister(definition.ListenPath, handler, definition.Methods, route.handlers)
+	p.doRegister(definition.ListenPath, handler.ServeHTTP, definition.Methods, route.inbound)
 	return nil
 }
 
-func (p *Register) doRegister(listenPath string, handler http.HandlerFunc, methods []string, handlers []router.Constructor) {
+func (p *Register) createDirector(proxyDefinition *Definition) func(req *http.Request) {
+	return func(req *http.Request) {
+		target, _ := url.Parse(proxyDefinition.UpstreamURL)
+		targetQuery := target.RawQuery
+
+		req.URL.Scheme = target.Scheme
+		req.URL.Host = target.Host
+		path := target.Path
+
+		if proxyDefinition.AppendPath {
+			log.Debug("Appending listen path to the target url")
+			path = singleJoiningSlash(target.Path, req.URL.Path)
+		}
+
+		if proxyDefinition.StripPath {
+			path = singleJoiningSlash(target.Path, req.URL.Path)
+			matcher := router.NewListenPathMatcher()
+			listenPath := matcher.Extract(proxyDefinition.ListenPath)
+
+			log.Debugf("Stripping listen path: %s", listenPath)
```

Comment:
do not use `log.<level>f()` -> `log.WithField(...).WithFields(...).<level>("<Static message>")`

in this case it is going to be `log.WithField("path", listenPath).Debug("Stripping listen path")`